packages/ozone/src/api/moderator/deleteUser.ts

```diff
@@ -0,0 +1,27 @@
+import { AuthRequiredError } from '@atproto/xrpc-server'
+import { Server } from '../../lexicon'
+import AppContext from '../../context'
+
+export default function (server: Server, ctx: AppContext) {
+  server.tools.ozone.moderator.deleteUser({
+    auth: ctx.authVerifier.modOrAdminToken,
+    handler: async ({ input, auth }) => {
+      const access = auth.credentials
+      const db = ctx.db
+      const { did } = input.body
+
+      if (!access.isAdmin) {
+        throw new AuthRequiredError(
+          'Must be an admin to delete a moderator user',
+        )
+      }
+      const moderatorService = ctx.moderatorService(db)
+
+      const canDelete = await moderatorService.canDelete(did)
+
+      if (canDelete) {
+        await moderatorService.delete(did)
+      }
```

Comment:
Since `canDelete()` can only return `true` or fail, I'd recommend renaming it to something like `assertCanDelete()` and not returning anything from it.

Another trick for these kinds of things, though, is to create a transaction, perform the deletion, check a post condition, and roll the postcondition back if it fails.  It can protect from races more effectively than checking a precondition.  Roughly:

```ts
await db.transaction(async (txn) => {
  const moderatorTxn = ctx.moderatorService(txn)
  await moderatorTxn.delete(did)
  await moderatorTxn.assertAdminExists() // throws if there are no admins, which causes transaction to rollback
})
```